FileIO/TetGenInterface.cpp

```diff
@@ -132,7 +132,7 @@ bool TetGenInterface::parseSmeshFacets(std::ifstream &input,
 	const unsigned offset = (_zero_based_idx) ? 0 : 1;
 	std::vector<std::size_t> idx_map;
 
-	for (std::size_t k(0); k<nFacets && !input.fail(); k++)
+	for (int k(0); k<nFacets && !input.fail(); k++)
```

Comment:
I tried to avoid cast. 

The problem of this specific code was that the variable k was used to control the number of read faceset which should be obvously not negative. But k was incremented in every iteration, even if not a faceset was read. I substituted the for loop by a while loop and only increment k when a faceset was read. I hope this fixes the problem.